lib/datadog/appsec/contrib/devise/event_information.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module Contrib
+      module Devise
+        # Module to extract event information from the resource
+        module EventInformation
+          UUID_REGEX = /^\h{8}-\h{4}-\h{4}-\h{4}-\h{12}$/.freeze
+
+          SAFE_MODE = 'safe'
+          EXTENDED_MODE = 'extended'
+
+          def self.extract(resource, mode)
+            event = {}
+
+            return event unless resource
+
+            resource_id = resource.id
+
+            case mode
+            when EXTENDED_MODE
+              resource_email = resource.email
+              resource_username = resource.username
+
+              event[:id] = resource_id if resource_id
+              event[:email] = resource_email if resource_email
+              event[:username] = resource_username if resource_username
+            when SAFE_MODE
+              event[:id] = resource_id if resource_id && resource_id.to_s =~ UUID_REGEX
```

Comment:
Certainly not for this PR but there is an opportunity here to define matchers (e.g as procs) and iterate over them, in order to easily extend the list of known safe bits of info.

Reply:
Also, for my understanding, wasn't there a thing specified where known unsafe bits of info would get hashed to pseudonymous information?

Reply:
> Also, for my understanding, wasn't there a thing specified where known unsafe bits of info would get hashed to pseudonymous information?

In the end for simplify we decided not to do it on the first iteration